builtin/logical/aws/path_static_creds.go

```diff
@@ -0,0 +1,99 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+const (
+	pathStaticCreds = "static-creds"
+
+	paramAccessKeyID      = "access_key_id"
+	paramSecretsAccessKey = "secret_access_key"
+)
+
+type awsCredentials struct {
+	AccessKeyID     string `json:"access_key_id" structs:"access_key_id" mapstructure:"access_key_id"`
+	SecretAccessKey string `json:"secret_access_key" structs:"secret_access_key" mapstructure:"secret_access_key"`
+}
+
+func pathStaticCredentials(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: fmt.Sprintf("%s/%s", pathStaticCreds, framework.GenericNameWithAtRegex(paramRoleName)),
+		Fields: map[string]*framework.FieldSchema{
+			paramRoleName: {
+				Type:        framework.TypeString,
+				Description: descRoleName,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback: b.pathStaticCredsRead,
+				Responses: map[int][]framework.Response{
+					http.StatusOK: {{
+						Description: http.StatusText(http.StatusOK),
+						Fields: map[string]*framework.FieldSchema{
+							paramAccessKeyID: {
+								Type:        framework.TypeString,
+								Description: descAccessKeyID,
+							},
+							paramSecretsAccessKey: {
+								Type:        framework.TypeString,
+								Description: descSecretAccessKey,
+							},
+						},
+					}},
+				},
+			},
+		},
+
+		HelpSynopsis:    pathStaticCredsHelpSyn,
+		HelpDescription: pathStaticCredsHelpDesc,
+	}
+}
+
+func (b *backend) pathStaticCredsRead(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	roleName, ok := data.GetOk(paramRoleName)
+	if !ok {
+		return nil, fmt.Errorf("missing '%s' parameter", paramRoleName)
+	}
+
+	entry, err := req.Storage.Get(ctx, formatCredsStoragePath(roleName.(string)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to read credentials for role '%q': %w", roleName, err)
```

Comment:
I do this so often, I need goland to have a lint for it

Reply:
To piggyback this, there's a couple places in the PR where you used `'%s'` separately. Thought it'd be easier to add that here rather than have individual comments on each one:

- [1](https://github.com/hashicorp/vault/pull/20536/files#diff-a40fc5c8db35ad9d98314bf847c7abd346408affb380b9cb2f7c6b2b813072ecR64)
- [2](https://github.com/hashicorp/vault/pull/20536/files#diff-2e2d943c34b628d1bbbc600d2fb67141e5b31db35433e2bd67d55c4ff45cb1ffR101)
- [3](https://github.com/hashicorp/vault/pull/20536/files#diff-2e2d943c34b628d1bbbc600d2fb67141e5b31db35433e2bd67d55c4ff45cb1ffR136)
- [4](https://github.com/hashicorp/vault/pull/20536/files#diff-2e2d943c34b628d1bbbc600d2fb67141e5b31db35433e2bd67d55c4ff45cb1ffR146)
- [5](https://github.com/hashicorp/vault/pull/20536/files#diff-2e2d943c34b628d1bbbc600d2fb67141e5b31db35433e2bd67d55c4ff45cb1ffR202)
- [6](https://github.com/hashicorp/vault/pull/20536/files#diff-db390cf464a714890c1014a50638adbc7099ce34f32a4b6cb70968185bac2fbfR91)
- [7](https://github.com/hashicorp/vault/pull/20536/files#diff-db390cf464a714890c1014a50638adbc7099ce34f32a4b6cb70968185bac2fbfR114)
- [8](https://github.com/hashicorp/vault/pull/20536/files#diff-db390cf464a714890c1014a50638adbc7099ce34f32a4b6cb70968185bac2fbfR123)

Reply:
done!